### PR TITLE
Improve the situation of Drag-and-Drop on X11

### PR DESCRIPTION
**Pull request by reduz**
Improves #38312 does not fix it

### CODE REVIEW

File: platform/osx/display_server_osx.mm [hunk 1]
```diff
@@ -3374,6 +3374,8 @@ void _update_keyboard_layouts() {
 }
 
 DisplayServer::WindowID DisplayServerOSX::get_window_at_screen_position(const Point2i &p_position) const {
+#warning This is an incorrect implementation, if windows overlap, it should return the topmost visible one or none if occluded by a foreign window
```
Comment (bruvzg) [inline platform/osx/display_server_osx.mm:3377]:
```obj-c++
	Point2i position = p_position;
	position.y *= -1;
	position += _get_screens_origin();
	position /= screen_get_max_scale();

	NSInteger wnum = [NSWindow windowNumberAtPoint:NSMakePoint(position.x, position.y) belowWindowWithWindowNumber:0 /*topmost*/];
	for (Map<WindowID, WindowData>::Element *E = windows.front(); E; E = E->next()) {
		if ([E->get().window_object windowNumber] == wnum) {
			return E->key();
		}
	}
	return INVALID_WINDOW_ID;
```

Comment (akien-mga) [reply platform/osx/display_server_osx.mm:3377]: I suggested that @reduz amends this in this commit but he forgot, so let's merge and you can make a fixup PR.

**Comment by akien-mga** [comment]
Thanks!

File: platform/osx/display_server_osx.mm (hunk 1 above)
Comment (reduz) [reply platform/osx/display_server_osx.mm:3377]: ah I forgot about this, please feel free to make it a separate PR

Comment (bruvzg) [reply platform/osx/display_server_osx.mm:3377]: I have not removed it from #40084 when rebasing it after this PR, so relevant code already get merged with it.

**Pull request merged**